src/manage.c

```diff
@@ -3614,11 +3615,13 @@ handle_osp_scan (task_t task, report_t report, const char *scan_id)
               osp_scan_status = get_osp_scan_status (scan_id, host, port,
                                                      ca_pub, key_pub, key_priv);
 
-              if (osp_scan_status == OSP_SCAN_STATUS_PENDING)
+              if (osp_scan_status == OSP_SCAN_STATUS_PENDING
+                  && pending_status_updated == FALSE)
```

Comment:
I think it would be clearer if pending_status_updated was checked in a new ````if```` inside the osp_scan_status ````if````.  Because if status is Pending then we don't need to run the ````else```` branch.